src/hotspot/share/opto/vectornode.hpp

```diff
@@ -854,6 +862,24 @@ class VectorToMaskNode : public VectorMaskNode {
   virtual Node* Identity(PhaseGVN* phase);
 };
 
+// Vector compare node with a TypeVMask bottom_type. It is specially generated for platforms
+// that have mask hardware feature. The main difference with "VectorMaskCmpNode" is that this
+// is a kind of mask node, while "VectorMaskCmpNode" is a vector node.
+class VectorCmpMaskGenNode : public VectorMaskNode {
+ public:
+  VectorCmpMaskGenNode(Node* in1, Node* in2, ConINode* predicate_node, const TypeVMask* vmask_type) :
```

Comment:
VectorMaskCmpNode compares two vectors and produces comparison results into a vector (for targets which does not supports predicated vectors) or a predicate vector/register.  Can you kindly elaborate the need for this new node.

Reply:
`VectorCmpMaskGen` is a kind of mask IR which generates predicate register.  The same with `VectorToMask`, we defined a series of mask IRs that are different with the original vector IRs.